plugins/cpp/parser/src/clangastvisitor.h

```diff
@@ -514,6 +519,28 @@ class ClangASTVisitor : public clang::RecursiveASTVisitor<ClangASTVisitor>
           friendship->theFriend = util::fnvHash(getUSR(friendDecl));
         }
       }
+
+      // --- Destructor --- //
+
+      void* ref = crd;
```

Comment:
why are you doing this? Is this variable used anywhere else later? It's really shocking to so violently get rid of type information...!

Reply:
The variable ref has type void* because the clangPtr_ parameter of addDestructorUsage that we pass it to also expects void*. And that is so because it is then further passed on to insertToCache, which also expects void*. So in essence, I just matched the type of parameters to an already existing pattern of expecting void* in the end at insertToCache. Whether or not this is a good pattern to begin with is of course up for debate, but I did not want to change the signature of insertToCache just for this, without fully knowing why we chose void* there in the first place.

Reply:
As we discussed on the weekly meeting, even in that case this type conversion should be done only when passing the argument to the `insertToCache` function. 
Don't loose tye information everywhere, just because one function won't need it!

Explicit conversion won't even be required, see other usages of the `insertToCache` function.

Reply:
OK, I have replaced the void* type of ref with the most derived common ancestor of the two types of AST nodes we put into it (see 2d56647ffe4d950ebc1b2a25175b37b78843eb22).